### PR TITLE
Shared ts migration

### PR DESCRIPTION
**Pull request by acelaya**
First batch of TypeScript migrations in `src/shared`. More coming next.

> I did not migrate all at once because the `injector` has a bunch of changes, type guards, etc., so I didn't want to make it harder to review.

### CODE REVIEW

File: src/shared/frame-error-capture.ts [hunk 1]
```diff
@@ -50,11 +44,8 @@ function serializeError(err) {
 
 /**
  * Convert error data serialized by {@link serializeError} back into an Error.
- *
- * @param {ErrorData} data
- * @return {Error}
  */
-function deserializeError(data) {
+function deserializeError(data: ErrorData): ErrorData {
```
Comment (robertknight) [inline src/shared/frame-error-capture.ts:48]: The return type here has been changed from `Error` to `ErrorData`.

File: src/shared/injector.ts [hunk 2]
```diff
@@ -1,22 +1,38 @@
...
- * @prop {Function} [class] - A class that should be instantiated to create the object
- * @prop {Function} [factory] - Function that should be called to create the object
- */
+type ValueProvider = { value: unknown };
 
-/**
- * @param {Provider} provider
- */
-function isValidProvider(provider) {
+type WithConstructor = {
```
Comment (robertknight) [inline src/shared/injector.ts:3]: Could alternately have called this `Constructible`. The current name is clear enough though.

**Review by robertknight — APPROVED** [review]
Found one function where the return type looks to have been changed inadvertently. Otherwise looks good 👍

File: src/shared/injector.ts (hunk 2 above)
Comment (acelaya) [reply src/shared/injector.ts:3]: I tried to come up with the `Constructible` word, but my English skills were not there yet 😂 

I like it better. Will change it.

**Pull request merged**